### ISSUE TITLE
Rule MA0115 - Unknown component parameter not working for third part packages

### ISSUE THREAD

**Issue by gabephudson** [meziantou/Meziantou.Analyzer#717]
### Version of the Meziantou.Analyzer NuGet package

2.0.150

### Rule Identifier

MA0115

### Target Framework

net8.0

### C# Language version

C# 12

### Description

When using a 3rd party NuGet Blazor package  (in this case, MudBlazor), this compiler warning does not occur for components in the third-party package. (Note, it does appear for internal components created in the project).

### Reproduction Steps

1. Create a new Blazor Web Project (.net 8) (with sample pages) in VS
2. Install MudBlazor NuGet package
3. Install Meziantou.Analyze NuGet package

4. In Home.Razor, add...
`<MudBlazor.MudButton InvalidParameter="abc">Test</MudBlazor.MudButton>`
`<Weather InvalidParamer="abc" />`

5. Compile solution.

Note that the MA0115 error is not present in the warnings for the Button, even though an invalid parameter is included in the component definition. It is, however, present for the Weather component.

### Other information

I am new to his package, so pardon my ignorance if this is just user error. Thank you for the wonder package!

**Comment by meziantou** [comment]
The issue is that the MudBlazor component allows any attribute because the base component has the `UserAttributes` property

````c#
[Parameter(CaptureUnmatchedValues = true)]
public Dictionary<string, object?> UserAttributes { get; set; } = new Dictionary<string, object?>();
````

I wonder if the rule should detect attribute starting with an uppercase letter that doesn't match a property. Most html attribute are lowercase. Maybe it would be a better heuristic to support this case.

**Comment by gabephudson** [comment]
Thanks for looking into this! Yeah, many Blazor UI components will allow this so they can pass through standard/non-standard HTML attributes.

As you said, these are typically lowercase where component parameters are often uppercase (by convention). That sounds very reasonable to me, although it is not bullet proof.

Would love to be able to see these flagged in this case, even if it's a separate rule that can be disabled for other's that don't want it flagged. I think this will be needed for any UI Blazor frameworks (which is what many NuGet Blazor packages are).

**Comment by meziantou** [comment]
The latest version should report unmatched pascal-case parameters

**Comment by gabephudson** [comment]
I can confirm this change now highlights MA0115 for this library. This is extremely useful (don't know why it's not part of the standard warnings). Thank you so much for the quick tweak and this wonderful set of analyzers!

**Comment by NaserParhizkar** [comment]
I would like to use Blazor parameters with something like [] same as Angular to be sure when I change a parameter name in the future, I could trace problem by use of editor and by use of compile errors. I see, only Angular could recognize and throw error when a parameter name change, even React can not detect parameter name changes.

**Comment by meziantou** [comment]
@NaserParhizkar If this is a feature, can you open another issue? Also, could you give an example of the pattern you want the rule to support?

**Comment by NaserParhizkar** [comment]
@meziantou Your tools (Meziantou.Analyzer) is good and resolve my question, but unfortunately when I click on warning ```
message  'Severity	Code	Description	Project	File	Line	Suppression State	Details
Warning (active)	MA0115	The parameter 'name' does not exist on component 'BlazorApp23.Components.Test.MyTest' (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0115.md)	BlazorApp23	C:\Users\Asus\source\repos\BlazorApp23\BlazorApp23\obj\Debug\net8.0\Microsoft.CodeAnalysis.Razor.Compiler\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\Components_Pages_Home_razor.g.cs	100		
```
'  
I expect to redirect to page I use the component with wrong property name in order to correct it.

**Comment by meziantou** [comment]
I think this is a limitation of Roslyn. MA0115 analyze the code generated by Razor. So, it doesn't analyze the razor file but a cs file (e.g. `Components_Pages_Home_razor.g.cs`). 
Razor uses the `#line` directive to do the mapping between the cs and razor file, but I don't know if Roslyn use it when double-clicking on the error.

**Comment by NaserParhizkar** [comment]
But other warnings like null reference warning, let the developers to show warning line of code by double click on it in visual studio  2022. 
You can check it yourself.
